### DISCUSSION TITLE
How to install Windows 10 system?

### DISCUSSION THREAD

**Discussion by ZhiyuanYuanNJ** [ipxe/ipxe#1266] [Q&A]
I have an ISO CD file that has been tested for a perfect installation of the Windows 10 system in a virtual machine. I extracted it and wrote some of the files into the IPXE installation script. Unfortunately, I received an error message:
READ BOOTX64.EFI
Could not load BOOTX64.EFI : 0x800000000003

**Here is my ipxe script：**
```shell
#!ipxe
kernel wimboot
initrd boot/bcd bcd
initrd boot/boot.sdi boot.sdi
initrd sources/boot.wim boot.wim
boot
``` 
The contents of the CD decompression file are as follows:
![image](https://github.com/user-attachments/assets/02433e1b-821d-4b11-a9b8-be7e80da1ccb)
**The winboot file was manually added by me.**

**Comment by NiKiZe** [comment]
Your boot.wim is missing `bootx64.efi`, the original boot.wim from official installation iso has this, WAIK created media might not, in that case you need to supply it as initrd.

Usually you don't need `boot.sdi` or `BCD` as these are internal or extracted from `boot.wim`

**Comment by ZhiyuanYuanNJ** [comment]
I found a bootx64.efi file and added it, but now ipxe is starting to prompt missing  ```\windows\system32\boot\winload.efi``` :)
Where can I obtain a complete ``` boot.wim ``` ?

**Reply by NiKiZe** [reply]
It's best to be precise about what stage is in process when you get a message, and it isn't iPXE. Is it windows bootloader, or wimboot, the process is the same, grab the files missing in `boot.wim` from your ISO.

The full boot.wim is found in the original installation media ISO.

**Reply by ZhiyuanYuanNJ** [reply]
> It's best to be precise about what stage is in process when you get a message, and it isn't iPXE. Is it windows bootloader, or wimboot, the process is the same, grab the files missing in `boot.wim` from your ISO.
> 
> The full boot.wim is found in the original installation media ISO.

I downloaded the win10(22H2) operating system from Microsoft's official website, unzipped the IOS file, and wrote this script on the tftp server:
```
#!ipxe
kernel wimboot
initrd boot/bcd bcd
initrd boot/boot.sdi boot.sdi
initrd sources/boot.wim boot.wim
boot
```
When ipxe shows that ```boot.wim``` has been downloaded, and then it stuck at the following interface:
![c4474c2e328a7c70becd631a945ed78](https://github.com/user-attachments/assets/4de71685-333f-4156-a455-40a58c7bf8b0)

What causes the problem?

**Reply by NiKiZe** [reply]
That should work, but remove BCD and boot.sdi lines, since they are not needed.

Please try some other hardware or VM.

At least, with a hang, you are past the previous issue with missing files.

The goal here is to get one working example, that you then can use to compare your non working setup/.wim to

**Reply by ZhiyuanYuanNJ** [reply]
Thank you for your help. Currently, I am not stuck in the above interface, but I am prompted that the CD/ROM driver is missing, like this:
![image](https://github.com/user-attachments/assets/19ddbf81-f2ee-4745-8d36-8e6091f71da0)
I have checked using the Dism++ tool and found that the ```boot.wim``` contains the driver program. At the same time, such errors have also appeared in the Win7 system.
Then I referred to [https://ipxe.org/howto/winpe](https://ipxe.org/howto/winpe) added ini and bat files, but it didn't work.

**Reply by NiKiZe** [reply]
Yes, because it can't find the sources path on the installation media. Now you need to create or modify boot.wim to have it work for your use case.

The first goal was to have a working boot.wim that you could compare your non working boot.wim to.

You could use the way I describe in https://github.com/NiKiZe/wimboot-install/tree/main/win10 (you can find the iPXE script a level up)
But it all depends on your end goal.

**Reply by ZhiyuanYuanNJ** [reply]
Got it, where can I get the  a working boot.wim.

**Reply by ZhiyuanYuanNJ** [reply]
And what I don't understand is that when I start from ISO, there won't be an error message indicating a missing driver. 
They all use the same boot.wim,  so why does starting through IPXE cause that error?

**Reply by NiKiZe** [reply]
You create your own boot.wim as explained on the page I linked (or any other place)

The "driver error" is because it can't find any storage device that contains the windows installation files, so it assumes you are lacking drivers for that storage.

So are you going to install windows? If so where will the installation files be located? As you can see if you look around a bit in the repo I linked, it uses `net use` to mount a network share.

**Reply by ZhiyuanYuanNJ** [reply]
Yes, I want to install Windows 10 system. I have set up an TFTP server in the local area network. I noticed that the script in the repository you linked seems to have used "initrd" to link the driver files. Should I do the same?

**Reply by NiKiZe** [reply]
It uses `initrd` for network drivers, if needed.
It also uses `initrd` for `startnet`

I will not give step by step instructions here, there is plenty of those already.
